assets/javascripts/discourse/models/filter-rule.js.es6

```diff
@@ -1,17 +1,22 @@
 import RestModel from 'discourse/models/rest';
 
 export default RestModel.extend({
-  category_id: -1,
+  category_id: '',
   channel: '',
   filter: null,
 
   category: function() {
     var id = this.get('category_id');
 
-    if (id === "0")
-      return Discourse.Category.create({ name: 'All Categories', id: 0 });
-    else {
-      return Discourse.Category.findById(id) || { id: id, name: 'Deleted Category' };
+    switch (id) {
+      case '*':
+        return Discourse.Category.create({ name: I18n.t('slack.choose.all_categories'), id: '*' });
+        break;
+      case '':
```

Comment:
I don't quite follow here. If someone chooses "All categories" and filters further by tags. We should only send topics that matches the tags in the filter and not ignore the tags because all categories is selected.